File: app/src/main/java/com/daemonw/deviceinfo/ui/main/BaseViewModel.java
package com.daemonw.deviceinfo.ui.main;

import android.content.Context;

import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public abstract class BaseViewModel<T> extends ViewModel {

    private MutableLiveData<T> info =new  MutableLiveData<T>();

    public MutableLiveData<T> get(){
        return info;
    }

    public void setValue(T info) {
        this.info.setValue(info);;
    }

    public T getValue(){
        return this.info.getValue();
    }

    abstract public BaseViewModel load(Context context);
}

File: app/src/main/java/com/daemonw/deviceinfo/ui/main/BaseViewModel.kt
package com.daemonw.deviceinfo.ui.main

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

abstract class BaseViewModel<T> : ViewModel() {
    private val info: MutableLiveData<T> = MutableLiveData()

    fun get(): MutableLiveData<T> {
        return info
    }

    fun setValue(info: T) {
        this.info.value = info
    }

    fun getValue(): T {
        return this.info.value as T
    }

    abstract fun load(): T
}